docs/src/faq.md

```diff
@@ -0,0 +1,402 @@
+```@meta
+CurrentModule = Enzyme
+DocTestSetup = quote
+    using Enzyme
+end
+```
+
+# Frequently asked questions
+
+## Implementing pullbacks
+
+In combined reverse mode, Enzyme's [`autodiff`](@ref) function can only handle functions with scalar output (this is not true for split reverse mode, aka `autodiff_thunk`).
+To implement pullbacks (back-propagation of gradients/tangents) for array-valued functions, use a mutating function that returns `nothing` and stores it's result in one of the arguments, which must be passed wrapped in a [`Duplicated`](@ref).
```

Comment:
Maybe add However, regardless of AD mode, it often much more efficient to use mutation, both for your original code, as well as the derivative. To implement pullbacks ...., consider using a mutating...

Reply:
done